titus-server-master/src/main/java/com/netflix/titus/master/eviction/service/DefaultEvictionOperations.java

```diff
@@ -16,401 +16,132 @@
-                                    .withNumberOfTokensPerInterval(2)
-                                    .build()
-                            )
-                            .build()
-            )
-            .build();
-
-    private static final SystemDisruptionBudget DEFAULT_CAPACITY_GROUP_DISRUPTION_BUDGET = SystemDisruptionBudget.newBuilder()
-            .withReference(Reference.global())
-            .withTokenBucketDescriptor(
-                    TokenBucketPolicy.newBuilder()
-                            .withCapacity(8)
-                            .withInitialNumberOfTokens(8)
-                            .withRefillPolicy(FixedIntervalTokenBucketRefillPolicy.newBuilder()
-                                    .withIntervalMs(4_000)
-                                    .withNumberOfTokensPerInterval(1)
-                                    .build()
-                            )
-                            .build()
-            )
-            .build();
-
-    private static final long QUOTA_UPDATE_INTERVAL_MS = 100;
-
     private final V3JobOperations jobOperations;
     private final ApplicationSlaManagementService capacityGroupService;
+    private final TitusQuotasManager quotaManager;
+    private final TitusRuntime titusRuntime;
+    private final Scheduler scheduler;
+    private final EvictionServiceConfiguration configuration;
 
-    private final TokenBucket globalTokenBucket = newTokenBucket("global", GLOBAL_DISRUPTION_BUDGET);
-
-    private final ConcurrentMap<Tier, TokenBucket> tierTokenBuckets = new ConcurrentHashMap<>();
-
-    private final ConcurrentMap<String, TokenBucket> capacityGroupTokenBuckets = new ConcurrentHashMap<>();
-
-    private final Object quotaLock = new Object();
-
-    private final FluxProcessor<EvictionEvent, EvictionEvent> eventSubject = EmitterProcessor.create();
-    private final Flux<EvictionEvent> eventObservable = ReactorExt.protectFromMissingExceptionHandlers(eventSubject, logger);
-
-    private final ConcurrentMap<Reference, Long> latestQuotaUpdates = new ConcurrentHashMap<>();
-    private Subscription eventEmitterSubscription;
+    private QuotaEventEmitter quotEventEmitter;
+    private TaskTerminationExecutor taskTerminationExecutor;
 
     @Inject
-    public DefaultEvictionOperations(V3JobOperations jobOperations,
+    public DefaultEvictionOperations(EvictionServiceConfiguration configuration,
+                                     V3JobOperations jobOperations,
                                      ApplicationSlaManagementService capacityGroupService,
+                                     TitusQuotasManager quotaManager,
+                                     TitusRuntime titusRuntime,
                                      ManagementSubsystemInitializer capacityGroupServiceInitializer) { // To enforce correct initialization order
+        this(configuration, jobOperations, capacityGroupService, quotaManager, titusRuntime, Schedulers.parallel());
```

Comment:
use a dedicated thread instead of `Schedulers.parallel()`?

Reply:
Done